packages/core/contracts/financial-templates/common/FundingRateApplier.sol

```diff
@@ -117,27 +226,27 @@ abstract contract FundingRateApplier is Testable, Lockable {
         }
 
         uint256 currentTime = getCurrentTime();
-        uint256 paymentPeriod = currentTime.sub(lastUpdateTime);
+        uint256 paymentPeriod = currentTime.sub(fundingRate.applicationTime);
```

Comment:
Where do you set `applicationTime`, I assume it should be in `_getLatestFundingRate`?

Reply:
It gets set at the end of this method (it basically just replaced lastUpdateTime).

This did make me realize that we don't initialize it in the constructor to the current time, which causes the first funding rate to be over a very long period. This didn't really have much of an impact because the funding rate starts at 0, but it's probably good to set it just to avoid any strange edge cases. So I'm adding a line to initialize it as it was being done before.

Reply:
> This did make me realize that we don't initialize it in the constructor to the current time, which causes the first funding rate to be over a very long period. This didn't really have much of an impact because the funding rate starts at 0, but it's probably good to set it just to avoid any strange edge cases. So I'm adding a line to initialize it as it was being done before.

Yes actually I do this exact same logic with the FundingRateStore and attempt to set it to the current time as soon as possible [here](https://github.com/UMAprotocol/protocol/blob/master/packages/core/contracts/financial-templates/funding-rate-store/implementation/FundingRateStore.sol#L190).

Reply:
> It gets set at the end of this method (it basically just replaced lastUpdateTime). Do you think it should be updated elsewhere?

My original question came up because I couldn't find the `fundingRate.applicationTime = currentTime` line but I see iw not at the end.